modules/mono/editor/Godot.NET.Sdk/Godot.SourceGenerators/Common.cs

```diff
@@ -67,6 +67,28 @@ TypeDeclarationSyntax outerTypeDeclSyntax
                 outerTypeDeclSyntax.SyntaxTree.FilePath));
         }
 
+        public static void ReportMultipleClassesInGodotScript(
+            GeneratorExecutionContext context,
+            ClassDeclarationSyntax cds, INamedTypeSymbol symbol
+        )
+        {
+            string message = $"Found multiple classes with the name '{symbol.Name}' in the same script file";
+
+            string description = $"{message}. A script file must only contain one class with a name that matches the file name.";
+
+            context.ReportDiagnostic(Diagnostic.Create(
+                new DiagnosticDescriptor(id: "GD0003",
+                    title: message,
+                    messageFormat: message,
+                    category: "Usage",
+                    DiagnosticSeverity.Error,
+                    isEnabledByDefault: true,
+                    description,
+                    helpLinkUri: string.Format(_helpLinkFormat, "GD0003")),
```

Comment:
- The link requires https://github.com/godotengine/godot-docs/pull/8973 to be merged so it doesn't open a page that doesn't exist.